bn-apps/memberships-management/membership-service/src/main/kotlin/com/r3/businessnetworks/membership/flows/bno/service/DatabaseService.kt

```diff
@@ -10,79 +11,88 @@ import net.corda.core.node.services.Vault
-        session.prepareStatement(nativeQuery).execute()
-    }
-
-    fun getMembership(member : Party, bno : Party, contract : String) : StateAndRef<MembershipState<Any>>? {
+    fun getMembership(member : Party, bno : Party) : StateAndRef<MembershipState<Any>>? {
         val criteria = QueryCriteria.VaultQueryCriteria(Vault.StateStatus.UNCONSUMED)
                 .and(memberCriteria(member))
                 .and(bnoCriteria(bno))
-        // filtering out the states validated by a correct contract
-        val states = serviceHub.vaultService.queryBy<MembershipState<Any>>(criteria).states.filterByContract(contract)
+        val states = serviceHub.vaultService.queryBy<MembershipState<Any>>(criteria).states
         return if (states.isEmpty()) null else (states.sortedBy { it.state.data.modified }.last())
     }
 
-    fun getAllMemberships(bno : Party, contract : String) : List<StateAndRef<MembershipState<Any>>> {
+    fun getAllMemberships(bno : Party) : List<StateAndRef<MembershipState<Any>>> {
         val criteria = QueryCriteria.VaultQueryCriteria(Vault.StateStatus.UNCONSUMED)
                 .and(bnoCriteria(bno))
-        return serviceHub.vaultService.queryBy<MembershipState<Any>>(criteria).states.filterByContract(contract)
+        return serviceHub.vaultService.queryBy<MembershipState<Any>>(criteria).states
     }
 
-    fun getActiveMemberships(bno : Party, contract : String)
-            = getAllMemberships(bno, contract).filter { it.state.data.isActive() }
+    fun getActiveMemberships(bno : Party)
+            = getAllMemberships(bno).filter { it.state.data.isActive() }
 
     /**
      * This method exists to prevent the same member from being able to request a membership multiple times, while their first membership transaction is being processed.
      * Pending membership request is created when a membership request arrives and is deleted when the membership transaction is finalised.
      * Any attempt to create a duplicate pending membership request would violate the DB constraint.
      */
     fun createPendingMembershipRequest(party : Party) {
-        val nativeQuery = """
-                insert into $PENDING_MEMBERSHIP_REQUESTS_TABLE ($PENDING_MEMBER_COLUMN)
-                values (?)
-            """
-        val session = serviceHub.jdbcSession()
-
-        val statement = session.prepareStatement(nativeQuery)
-        statement.setString(1, party.name.toString())
-        statement.executeUpdate()
+        serviceHub.withEntityManager {
+            persist(PersistentPendingMembershipRequest.from(party))
+            flush()
+        }
     }
 
-
     fun deletePendingMembershipRequest(party : Party) {
-        val nativeQuery = """
-                delete from $PENDING_MEMBERSHIP_REQUESTS_TABLE
-                where $PENDING_MEMBER_COLUMN = ?
+        serviceHub.withEntityManager {
+            val nativeQuery = """
+                delete from PersistentPendingMembershipRequest
```

Comment:
Consider `CriteriaBuilder`

Reply:
Pls see the above re: HQL. Renamed nativeQuery->hqlQuery